### DISCUSSION TITLE
Using GitHub Actions with an IP allow list

### DISCUSSION THREAD

**Discussion by braedongough** [community/community#27106] [Enterprise]
<p>We’ve recently migrated to Github Enterprise Cloud so that we can begin to leverage the IP allowlist.</p>
<p>It says in the <a href="https://docs.github.com/en/organizations/keeping-your-organization-secure/managing-allowed-ip-addresses-for-your-organization#using-github-actions-with-an-ip-allow-list" rel="noopener nofollow ugc">documentation</a> that we must use self-hosted runners now.</p>
<p>I’ve set up a test org that enforces an IP allowlist and have been able to successfully run a workflow using Github-hosted runners.</p>
<p>Is the above documentation still up-to-date? If it is true, could anoyone further elaborate on which features of actions won’t be available once we enforce an IP allowlist on our primary organization?</p>

**Comment by braedongough** [comment, marked as answer]
<p>I was able to confirm that the IP allowlist does in fact disallow Github-hosted runners from executing properly.</p>
<p>When attempting to run the checkout action, my workflow failed with a 403.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/e/cef03da1cb5a082658520379942eeb613ba0545b.png" data-download-href="/uploads/short-url/twF4u5Zf1xOycDqlg7ltASkeuOT.png?dl=1" title="Screen Shot 2021-06-01 at 15.07.53" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/38016720/181101718-4e2000e7-3eb1-4a3c-85dd-2c46407dcf8f.png" alt="Screen Shot 2021-06-01 at 15.07.53" data-base62-sha1="twF4u5Zf1xOycDqlg7ltASkeuOT" width="690" height="224" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/e/cef03da1cb5a082658520379942eeb613ba0545b_2_690x224.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/e/cef03da1cb5a082658520379942eeb613ba0545b_2_1035x336.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/c/e/cef03da1cb5a082658520379942eeb613ba0545b_2_1380x448.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/e/cef03da1cb5a082658520379942eeb613ba0545b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2021-06-01 at 15.07.53</span><span class="informations">1618×526 108 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by mickeygousset** [comment]
<p>You might be able to add the IP range for the hosted runners and it might work, but the problem is that IP range could change constantly.</p>

**Comment by braedongough** [comment]
<p>Yeah, that’s my current plan.</p>

**Comment by byrneh** [comment]
<p><a class="mention" href="/braedongough">@braedongough</a>,<br>
GitHub serves applications from multiple IP address ranges, which are available using the API <a href="https://api.github.com/meta" rel="noopener nofollow ugc">meta API endpoint</a></p>
<p>It looks a pretty <strong>ugly</strong> and vast range of possible GitHub IP addresses related to GitHub hosted runners and all IP address ranges are subject to change.<br>
Be aware they are shared so it would anyone can use an ACTIONS IP address as an attack vector if IP ranges added to your allow list, it would seem like it reduces the control effectiveness of your IP Allow list restriction you have configured.  You may also hit problems with the granularity and number of IP address ranges to add and number of entries permitted for IP Allow configuration</p>

**Comment by ps-jay** [comment]
@braedongough, you can leverage `actions/upload-artifact` and `actions/download-artifact` - as artifact storage is not subject to the IP allow list.

Have your self-hosted runners upload your code base, then use the GitHub hosted runners to do the build.

We've wrapped checkout and upload in an action for convenience: https://github.com/reecetech/transfer-action/
